libmscore/stafftypechange.cpp

```diff
@@ -108,9 +108,24 @@ void StaffTypeChange::draw(QPainter* painter) const
          lw, Qt::SolidLine, Qt::SquareCap, Qt::MiterJoin));
       painter->setBrush(Qt::NoBrush);
       painter->drawRect(0, 0, w, h);
-      QFont f("FreeSans", 12.0 * _spatium * MScore::pixelRatio / SPATIUM20);
-      painter->setFont(f);
-      painter->drawText(QRectF(0.0, 0.0, w, h), Qt::AlignCenter, QString("S"));
+
+      // draw simple preview icon
+      int lines = 5;
+      if (staffType()) {
+            if (staffType()->stemless())
+                  drawSymbol(SymId::noteheadBlack, painter, QPoint(w * 0.5 - 0.33 * _spatium, h * 0.5), 0.5);
+            if (staffType()->invisible())
+                  return;
+            else
+                  lines = qMin(staffType()->lines(),6);   // show up to 6 lines
+            }
+      qreal startY = 1.425 - lines * 0.175;               // show lines centered
```

Comment:
magic numbers?

Reply:
The (h)eight of the square is 2.5, so the center Y = 1.25.  
0.35 is the distance between lines I stipulated to fit up to 6 lines in the square. 0.175 is half this distance, which will be multiplied * the number of lines and substracted from center Y so the lines draw centered.
But... the first line should not substract, so it would actually be something like
startY = 1.25 - (lines - 1) * 0.35 / 2
Would that be more clear? Sorry I just made a quick math in my head while coding.

Reply:
Quick maths are perfect, but comments are okay too ;)

Reply:
You are right! Just improved (and optimized a little) the code.